python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -154,6 +159,120 @@ def repository_name(self) -> Optional[str]:
         return self._repository_name
 
 
+class MultiAssetSensorEvaluationContext(SensorEvaluationContext):
+    """The context object available as the argument to the evaluation function of a :py:class:`dagster.MultiAssetSensorDefinition`.
+
+    Users should not instantiate this object directly. To construct a
+    `MultiAssetSensorEvaluationContext` for testing purposes, use :py:func:`dagster.
+    build_multi_asset_sensor_context`.
+
+    Attributes:
+        instance_ref (Optional[InstanceRef]): The serialized instance configured to run the schedule
+        cursor (Optional[Mapping[AssetKey, str]]): The cursor, passed back from the last sensor evaluation via
+            the cursor attribute of SkipReason and RunRequest
+        last_completion_time (float): DEPRECATED The last time that the sensor was evaluated (UTC).
+        last_run_key (str): DEPRECATED The run key of the RunRequest most recently created by this
+            sensor. Use the preferred `cursor` attribute instead.
+        repository_name (Optional[str]): The name of the repository that the sensor belongs to.
+        instance (Optional[DagsterInstance]): The deserialized instance can also be passed in
+            directly (primarily useful in testing contexts).
+
+    Example:
+
+    .. code-block:: python
+
+        from dagster import multi_asset_sensor, MultiAssetSensorEvaluationContext
+
+        @multi_asset_sensor(asset_keys=[AssetKey("asset_1), AssetKey("asset_2)])
+        def the_sensor(context: MultiAssetSensorEvaluationContext):
+            ...
+
+    """
+
+    def __init__(
+        self,
+        instance_ref: Optional[InstanceRef],
+        last_completion_time: Optional[float],
+        last_run_key: Optional[str],
+        cursor: Optional[Mapping[AssetKey, str]],
+        repository_name: Optional[str],
+        asset_keys: Sequence[AssetKey],
+        instance: Optional[DagsterInstance] = None,
+    ):
+        self._asset_keys = asset_keys
+        self.cursor_has_been_updated = False
+
+        super(MultiAssetSensorEvaluationContext, self).__init__(None, None, None, None, None, None)
+
+        self._exit_stack = ExitStack()
+        self._instance_ref = check.opt_inst_param(instance_ref, "instance_ref", InstanceRef)
+        self._last_completion_time = check.opt_float_param(
+            last_completion_time, "last_completion_time"
+        )
+        self._last_run_key = check.opt_str_param(last_run_key, "last_run_key")
+        self._cursor = check.opt_dict_param(cursor, "cursor", key_type=AssetKey, value_type=str)
+        self._repository_name = check.opt_str_param(repository_name, "repository_name")
+        self._instance = check.opt_inst_param(instance, "instance", DagsterInstance)
+
+    def latest_materializations_by_key(self, asset_keys: Optional[Sequence[AssetKey]] = None):
```

Comment:
When I suggested the name `latest_materializations_by_key`, I was thinking that the values of the returned dictionary would be `AssetMaterialization` objects. I see the wisdom in returning `EventLogRecord` objects though, because then users can get the storage ID and timestamp.  Maybe a name like `latest_materialization_records_by_key` would be clearer in that case?

Reply:
Analogous naming feedback applies to the `materializations_for_key` method and the `materializations_by_key` arg to `advance_cursor`.